tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,793 @@
+    lut = {}
+    for k, v in self.path.items():  # k:v, <zipfile#>:<path of the zipfile>
+      with ZipFile(v, 'r') as zip:
+        for name in zip.namelist(
+        ):  # name, "Image_pngs/<seriesFolder>/<xxx>.png"
+          seriesFolder = name.split('/')[1]
+          lut[seriesFolder] = v
+    self._lut = lut
+
+
+class FileNameUtils():
+  """Helper class to parse fileName from the annotation file
+
+  Attributes:
+    fileName: `str`, "<seriesFolder>_<%03d>.png"
+    _seriesFolder: `str`, "<seriesFolder>"
+    _sliceIdx: `int`, d
+    _sliceFileName: `str`, "<%03d>.png"
+    _filePath: `str`, "<seriesFolder>/<%03d>.png"
+  """
+  def __init__(self, fileName):
+    """Inits with a fileName"""
+    self.fileName = fileName
+    self._seriesFolder = None
+    self._sliceIdx = None
+    self._sliceFileName = None
+    self._filePath = None
+
+  @property
+  def seriesFolder(self):
+    if not self._seriesFolder:
+      self._parse_fname()
+    return self._seriesFolder
+
+  @property
+  def sliceIdx(self):
+    if not self._sliceIdx:
+      self._parse_fname()
+    return self._sliceIdx
+
+  @property
+  def sliceFileName(self):
+    if not self._sliceFileName:
+      self._sliceFileName = '%03d.png' % self.sliceIdx
+    return self._sliceFileName
+
+  @property
+  def filePath(self):
+    if not self._filePath:
+      self._filePath = os.path.join(self.seriesFolder,
+                                    self.sliceFileName)
+    return self._filePath
+
+  def _parse_fname(self):
+    """Parses fileName into _seriesFolder and _sliceIdx
+    """
+    self._sliceIdx = int(self.fileName.split('_')[-1][:-4])
+    self._seriesFolder = '_'.join(self.fileName.split('_')[:-1])
+
+  def get_fname(self, sliceIdx):
+    """Returns a fileName in annotated format
+
+    Args:
+      sliceIdx: `int`, index of a slice (the index should in valid slice_range)
+
+    Returns:
+      a `str` in "<seriesFolder>_<%03d>.png" format
+    """
+    return '_'.join([self.seriesFolder, '%03d.png' % sliceIdx])
+
+  def get_fname_list(self, start, end):
+    """Returns a list of names contructed by continuous indices
+
+    Args:
+      start: `int`, start point of continuous indices
+      end: `int`, end point of continuous indices (inclusive)
+
+    Returns:
+      a list of `str`
+    """
+    return [self.get_fname(s) for s in range(start, end + 1)]
+
+
+class AnnParser():
+  """Deeplesion Annotation Parser
+
+  Attributes:
+    ann_path: `str`, path of the annotation file
+    config: `tfds.core.BuilderConfig`, builder_config
+    _annTrain: `pandas.Dataframe`, annotations of train split
+    _annVal: `pandas.Dataframe`, annotations of validation split
+    _annTest: `pandas.Dataframe`, annotations of test split
+  """
+  def __init__(self, ann_path, config=None):
+    """Inits with path of the annotation file
+    """
+    self.ann_path = ann_path
+    self.config = config
+    self._annTrain = None
+    self._annVal = None
+    self._annTest = None
+
+  @property
```

Comment:
you could change these 3 properties into a single function.

(if you also change _annTrain, annVal and annTest into a dict)